crates/telio-wg/src/no_link_detection.rs

```diff
@@ -0,0 +1,522 @@
+use std::{collections::HashMap, time::Duration};
+use tokio::time::Instant;
+
+use telio_crypto::PublicKey;
+use telio_model::{
+    api_config::FeatureNoLinkDetection,
+    mesh::{LinkState, NodeState},
+};
+
+/// Default wireguard keepalive duration
+const WG_KEEPALIVE: Duration = Duration::from_secs(10);
+/// Default configurable rtt value (15 sec)
+const DEFAULT_CONFIGURABLE_RTT: Duration = Duration::from_secs(15);
+
+pub struct NoLinkDetection {
```

Comment:
This structure seems to serve a dual purpose, depending on whether no link detection is enabled or disabled. My suggestion is to have two private structures
```rust
struct NoLinkDetectionEnabled{
    ...
}
````
and 
```rust
struct NoLinkDetectionEnabled {
     ...
}
```
Then putting these into the common structure with public API:
```rust
pub enum NoLinkDetecction {
    Enabled(NoLinkDetectionEnabled),
    Disabled(NoLinkDetectionDisabled),
}
```
alternatively (I don't like this) to have a `Box<dyn NoLinkDetection>` where `NoLinkDetection` is a trait implemented by both private structures.
Why? Because it will save you a ton of state consistency checks in the methods like `if self.is_disabled()` and stuff.